backend/infrahub/graphql/queries/event.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from graphene import Field, Int, List, NonNull, ObjectType, String
+from infrahub_sdk.utils import extract_fields_first_node
+
+from infrahub.graphql.types.event import EventNodes
+from infrahub.task_manager.event import PrefectEvent
+
+if TYPE_CHECKING:
+    from graphql import GraphQLResolveInfo
+
+
+class Events(ObjectType):
+    edges = List(NonNull(EventNodes), required=True)
+    count = Int(required=True)
+
+    @staticmethod
+    async def resolve(
+        root: dict,  # pylint: disable=unused-argument
+        info: GraphQLResolveInfo,
+        limit: int = 10,
+        offset: int = 0,
+        ids: list[str] | None = None,
+        branch: str | None = None,
+        q: str | None = None,
+    ) -> dict[str, Any]:
+        # related_nodes = related_node__ids or []
+        ids = ids or []
+        return await Events.query(
+            info=info,
+            branch=branch,
+            limit=limit,
+            offset=offset,
+            q=q,
+            ids=ids,
+        )
+
+    @classmethod
+    async def query(
+        cls,
+        info: GraphQLResolveInfo,
+        q: str | None = None,
+        ids: list[str] | None = None,
+        branch: str | None = None,
+        limit: int | None = None,
+        offset: int | None = None,
+    ) -> dict[str, Any]:
+        fields = await extract_fields_first_node(info)
+
+        prefect_tasks = await PrefectEvent.query(
+            fields=fields,
+            q=q,
+            ids=ids,
+            branch=branch,
+            limit=limit,
+            offset=offset,
+        )
+        prefect_count = prefect_tasks.get("count", None)
+        return {
+            "count": prefect_count or 0,
```

Comment:
Instead of having `None` I'd suggest `prefect_count = prefect_tasks.get("count", 0). Perhaps even having it directly on the "count" key so that we don't have to create a variable for it here.